jupyterhub/templates/hub/secret.yaml

```diff
@@ -10,11 +10,15 @@ data:
   {{- if .Values.hub.cookieSecret }}
   hub.cookie-secret: {{ .Values.hub.cookieSecret | b64enc | quote }}
   {{- end }}
-  {{- range $key, $value := .Values.hub.services }}
-  {{- if $value.apiToken }}
-  services.token.{{$key}}: {{ $value.apiToken | b64enc | quote }}
-  {{- end }}
-  {{- end }}
   {{- if .Values.auth.state.enabled }}
   auth.state.crypto-key: {{ (required "Encryption key is required for auth state to be persisted!" .Values.auth.state.cryptoKey) | b64enc | quote }}
   {{- end }}
+  {{- $values := dict "hub" dict }}
+  {{- /* pluck only needed secret values, preserving values.yaml structure */ -}}
+  {{- $_ := set $values.hub "services" dict -}}
+  {{- range $key, $service := .Values.hub.services -}}
+    {{- if $service.apiToken -}}
+      {{- $_ := set $values.hub.services $key (pick $service "apiToken") -}}
+    {{- end -}}
+  {{- end }}
```

Comment:
See https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/941/files#r219466675

Reply:
Indeed, looks like there's nothing to change here.

Reply:
right-chomps removed.